### PR TITLE
Allow single name component repository names

### PR DESCRIPTION
**Pull request by dmcgowan**
Will allow images with a single name component, on the client side they may appear as <private repo>/myapp

### CODE REVIEW

File: api/v2/names.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ var RepositoryNameComponentAnchoredRegexp = regexp.MustCompile(`^` + RepositoryN
 
 // RepositoryNameRegexp builds on RepositoryNameComponentRegexp to allow 2 to
```
Comment (ncdc) [inline api/v2/names.go]: s/2/1 ??

Comment (stevvooe) [inline api/v2/names.go]: :+1:

**Comment by stevvooe** [comment]
Please add some testcases in routes_test.go and names_test.go.

**Comment by dmcgowan** [comment]
Had to update `RepositoryNameMinComponents` value as well to get unit tests passing.  Strangely I did not need to update that value to get a registry push to work.

**Comment by stevvooe** [comment]
@dmcgowan Unfortunately, the validation function is not yet wired up in the web application. We'll add that this week along with some refactoring.

**Comment by stevvooe** [comment]
LGTM!

**Pull request merged**